stdlib/sql/vertica.go

```diff
@@ -0,0 +1,170 @@
+		case []uint8:
+			fmt.Print("hack:", m.columnTypes[i])
+			// Hack for MySQL, might need to work with charset?
+			// Can't do boolean with MySQL - stores BOOLEANs as TINYINTs (0 or 1)
+			// No way to distinguish if intended int or bool
+			switch m.columnTypes[i] {
+			case flux.TInt:
+				newInt, err := UInt8ToInt64(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewInt(newInt)
+			case flux.TFloat:
+				newFloat, err := UInt8ToFloat(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewFloat(newFloat)
+			// This works, but you can also just just add the DSN parameter parseTime=true (see line 136)
+			case flux.TTime:
+				t, err := time.Parse(layout, string(col))
+				if err != nil {
+					fmt.Print(err)
+				}
+				row[i] = values.NewTime(values.ConvertTime(t))
+			default:
+				row[i] = values.NewString(string(col))
+			}
+		case nil:
+			row[i] = values.NewNull(flux.SemanticType(m.columnTypes[i]))
+		default:
+			execute.PanicUnknownType(flux.TInvalid)
+		}
+	}
+	return row, nil
+}
+
+func (m *VerticaRowReader) InitColumnNames(n []string) {
+	m.columnNames = n
+}
+
+func (m *VerticaRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	stringTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "INT", "INTEGER", "BIGINT", "SMALLINT", "TINYINT", "INT2", "INT4", "INT8", "SERIAL2", "SERIAL4", "SERIAL8":
+			stringTypes[i] = flux.TInt
+		case "FLOAT", "FLOAT4", "FLOAT8":
+			stringTypes[i] = flux.TFloat
+		case "DATE", "TIME", "TIMESTAMP":
+			stringTypes[i] = flux.TTime
+		case "BOOL":
+			stringTypes[i] = flux.TBool
+		case "TEXT", "VARCHAR", "VARBINARY":
+			stringTypes[i] = flux.TString
+		default:
+			stringTypes[i] = flux.TString
+		}
+	}
+	m.columnTypes = stringTypes
+}
+
+func (m *VerticaRowReader) ColumnNames() []string {
+	return m.columnNames
+}
+
+func (m *VerticaRowReader) ColumnTypes() []flux.ColType {
+	return m.columnTypes
+}
+
+func (m *VerticaRowReader) SetColumns(i []interface{}) {
+	m.columns = i
+}
+
+func (m *VerticaRowReader) Close() error {
+	if err := m.Cursor.Err(); err != nil {
+		return err
+	}
+	return m.Cursor.Close()
+}
+
+func NewVerticaRowReader(r *sql.Rows) (execute.RowReader, error) {
+	reader := &VerticaRowReader{
+		Cursor: r,
+	}
+	cols, err := r.Columns()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnNames(cols)
+
+	types, err := r.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnTypes(types)
+	return reader, nil
+}
+
+// VerticaTranslateColumn translates flux colTypes into their corresponding Vertica column type
+func VerticaColumnTranslateFunc() translationFunc {
+	c := map[string]string{
+		flux.TFloat.String():  "FLOAT",
+		flux.TInt.String():    "INTEGER",
+		flux.TUInt.String():   "INTEGER",
+		flux.TString.String(): "VARCHAR",
+		flux.TTime.String():   "TIMESTAMP",
+		flux.TBool.String():   "BOOL",
```

Comment:
In the `postgres` package, values with the flux type `TInt` and `TUInt` map to the SQL type `BIGINT`, and values with the flux type `TString` map to the SQL type `TEXT`. Just curious why that wouldn't work here. Does vertica not support those types?